packages/svelte-query/tsconfig.json

```diff
@@ -1,14 +1,28 @@
 {
-  "extends": "./.svelte-kit/tsconfig.json",
```

Comment:
should we now extend our `base.tsconfig.json` file? What I'd like to have is consistent typescript settings enabled that are not enabled by default, like:

```
"noUncheckedIndexedAccess": true,
```

our base config ensures that.

Reply:
I've tried this, but it seems to mess with `svelte-package`'s logic which determines where the `.d.ts` files get built to, and leads to the structure below. Is it okay if for now I copy the rules that don't break it from tsconfig.base.json?

![image](https://user-images.githubusercontent.com/1667261/211662472-d732bcf6-9a29-4d23-936d-9f8a567e62a1.png)

Reply:
yep 👍

Reply:
I got a bit closer to making this work with `tsconfig.base.json` - `tsc` reports no errors (but doesn't pass svelte files), while `svelte-check` gives the error below. The inconsistency between the two means `svelte-check` is probably not reading the config identically to `tsc` which is causing problems.

```
Error: File 'c:/Users/XYZ/Documents/GitHubProjects/tanstack-query/packages/query-core/src/index.ts' is not under 'rootDir' 'c:/Users/XYZ/Documents/GitHubProjects/tanstack-query/packages/svelte-query/src'. 'rootDir' is expected to contain all source files.
```

Reply:
Okay fairly certain this will not work for now, but the new rules are working :)